_posts/2022-03-07-changelog-166.markdown

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #166"
+date: 2022-03-07 10:00:00 +0300
+---
+
+
+## New Features
+
+* [#8446] Detect [E0049](https://doc.rust-lang.org/error-index.html#E0049) (Implementation has wrong number of type or const parameters)
+
+  <img src="/assets/posts/changelog-166/wrong_number_arguments.png" width="700px"/>
+
+## Fixes
+
+* [#8531] Take into account type arguments/parameters ordering and const generics in some intentions/fixes
+
+* [#7864] Correctly handle `+toolchain` at the beginning of Cargo command line
+
+* [#8578] Add `--respect-source-config` to `cargo vendor` command  to respect `[source]` config in `.cargo/config`.
+  This fixes the problem when using another source of cargo registry but still stuck in 'updating crates.io index' (by [@Kylin3216])
+
+* [#8524] Use `Self` when generating constructor in the `Generate Constructor` refactoring action (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-166/generate_constructor_self" w="700px" %}
+
+* [#7977] Render generic parameters for associated types in `Implement Members`
```

Comment:
Since this PR doesn't cover all cases, let's rephrase it to something like

```suggestion
* [#7977] Improve rendering of generic parameters for associated types in `Implement Members` quick-fix
```